### DISCUSSION TITLE
Global configuration for custom MDX components

### DISCUSSION THREAD

**Discussion by ohansemmanuel** [withastro/docs#3063] [Ideas]
Hey there 👋🏽 

Out of curiosity, Is there an ongoing consideration to make it possible to **configure custom MDX components for an Astro project in one place?** Without repetition across multiple pages/files? 

![https://docs.astro.build/en/guides/markdown-content/#custom-components-with-imported-mdx](https://user-images.githubusercontent.com/10930234/232403824-5b5ba8a5-c008-405d-a924-060adc6f95cc.png)

As it stands, it appears custom components have to be [defined per page](https://docs.astro.build/en/guides/markdown-content/#custom-components-with-imported-mdx) where they are consumed.

While one may arguably build a remark plugin / Astro integration to auto-import components, it might be nice to support configuring these in one place, if it doesn't already exist. Similar to how NextJS uses an [MDXProvider](https://nextjs.org/docs/advanced-features/using-mdx) in the app.js file. 

Potentially, these may work: 

- configure components in a layout component, and use this layout in every MDX file 
- expose a `markdown` configuration option in `astro.config.mjs` 
- some other clever idea, as I haven't given this deep thought

... 

Cheers

**Comment by yanthomasdev** [comment]
Hey @ohansemmanuel 👋, we don't have plans for global MDX components for now, according to [our public roadmap](https://github.com/orgs/withastro/projects/11/views/1).

The community package [astro-auto-import](https://github.com/delucis/astro-auto-import/tree/main/packages/astro-auto-import) is compatible with MDX and works similarly to what you proposed, I hope it can help you!

EDIT: Also, our recent Markdoc integration, which is a different file format/syntax, so I would not recommend migrating to it only for that, can do that: https://docs.astro.build/en/guides/integrations-guide/markdoc/#render-markdoc-tags-as-astro-components

**Reply by ohansemmanuel** [reply]
In that case, auto-importing suffices for now. Many thanks, @Yan-Thomas 🙏
